en/dev_setup/dev_env_windows_wsl.md

```diff
@@ -0,0 +1,161 @@
+   ```bash
+   bash ./PX4-Autopilot/Tools/setup/ubuntu.sh
+   ```
+   
+   :::note
+   This installs tools to build PX4 for Pixhawk, Gazebo and JMAVSim targets.
+   - You can use the `--no-nuttx` and `--no-sim-tools` options to omit the NuttX and/or simulation tools.
+   - Other Linux build targets are untested (you can try these by entering the appropriate commands in [Ubuntu Development Environment](../dev_setup/dev_env_linux_ubuntu.md) into the WSL shell.
+     Note that `git` is already installed in WSL.
+   
+1. Restart the "WSL computer" on completion.
+   ```
+   wsl --shutdown
+   ```
+   (or do `exit` and reopen the prompt).
+
+1. Execute command `wsl` to start WSL again and access the shell.
+1. Execute `cd ~/PX4-Autopilot` to switch to the PX4 repository in the WSL home folder.
+1. Execute `make px4_sitl` to build the PX4 SITL target and test your environment.
+
+
+### Opening a WSL shell
+
+To open the WSL shell (once WSL has been installed):
+
+1. Press the Windows **Start** key.
+1. Type "cmd" and press Enter to open a command prompt.
+1. Execute the command `wsl` to access the WSL shell.
+
+
+## Visual Studio Code Integration
+
+1. Download and install Visual Studio Code on Windows from https://code.visualstudio.com/
+2. Open _Visual Studio Code_.
+3. Install the extension called [Remote - WSL](https://marketplace.visualstudio.com/items?itemName=ms-vscode-remote.remote-wsl) (marketplace) 
+4. [Open a WSL shell](dev_env_windows_wsl.md#opening-a-wsl-shell)
+5. Switch to the PX4 folder: `cd ~/PX4-Autopilot`
+6. Open this folder with WSL integration in Visual Studio Code `code .`
+7. Now you have the full IDE integration with Terminal, make sure you always open the PX4 repository in the Remote WSL mode.
+   It will be available in your recently opened locations whenever you start VS Code.
+
+## QGroundControl
+
+You can run QGroundControl in either WSL or Windows to connect to the running simulation.
+If you need to [flash a flight control board](#flash-a-flight-control-board) with new firmware you can only do this from the QGroundControl for Windows.
+
+### QGroundControl in WSL
+
+The easiest way to set up and use QGroundControl is to download the Linux version into your WSL.
+This will automatically connect to a running simulation and allow you to monitor and control your vehicle(s).
+
+You will not be able to use it to install PX4 firmware because WSL does not allow access to serial devices.
+
+
+### QGroundcontrol on Windows
+
+Install QGroundControl on Windows if you want to be able to update hardware with firmware created within PX4.
+
+These steps describe how you can connect to the simulation running in the WSL.
+
+1. [Open a WSL shell](dev_env_windows_wsl.md#opening-a-wsl-shell)
+2. Check what IP address it received using the command `ip address` and copy it to clipboard by selecting the IP and using the shortcut [Ctrl] + [Shift] + [C].
```

Comment:
1. Does this ever change? I.e. do I have to get new address every time I run the WSL?
   If so, yuk. Might be better to suggest use QGC in WSL for flying and ANOTHER QGC in windows, just for uploading. Thoughts?
   
2. This is what I see. Is the IP address the one listed for Ethernet? 172.19.7.31/20 

```
1: lo: <LOOPBACK,UP,LOWER_UP> mtu 65536 qdisc noqueue state UNKNOWN group default qlen 1000
    link/loopback 00:00:00:00:00:00 brd 00:00:00:00:00:00
    inet 127.0.0.1/8 scope host lo
       valid_lft forever preferred_lft forever
    inet6 ::1/128 scope host
       valid_lft forever preferred_lft forever
2: bond0: <BROADCAST,MULTICAST,MASTER> mtu 1500 qdisc noop state DOWN group default qlen 1000
    link/ether 2e:0f:f9:cc:bd:77 brd ff:ff:ff:ff:ff:ff
3: dummy0: <BROADCAST,NOARP> mtu 1500 qdisc noop state DOWN group default qlen 1000
    link/ether ae:09:c7:b6:44:e7 brd ff:ff:ff:ff:ff:ff
4: tunl0@NONE: <NOARP> mtu 1480 qdisc noop state DOWN group default qlen 1000
    link/ipip 0.0.0.0 brd 0.0.0.0
5: sit0@NONE: <NOARP> mtu 1480 qdisc noop state DOWN group default qlen 1000
    link/sit 0.0.0.0 brd 0.0.0.0
6: eth0: <BROADCAST,MULTICAST,UP,LOWER_UP> mtu 1500 qdisc mq state UP group default qlen 1000
    link/ether 00:15:5d:be:1b:8f brd ff:ff:ff:ff:ff:ff
    inet 172.19.7.31/20 brd 172.19.15.255 scope global eth0
       valid_lft forever preferred_lft forever
    inet6 fe80::215:5dff:febe:1b8f/64 scope link
       valid_lft forever preferred_lft forever
```

Reply:
1. Every time WSL is restarted either because you reboot the computer or because you do that manually using `wsl --shutdown`. I know it's not great, not sure how this IP could be made static. I didn't find an easy solution.
2. Exactly `172.19.7.31` in your case.

Reply:
The problem here is that it changes every time - so you would have to update QGC every time. That's a real pain. I guess for now we document it, and think about possible alternatives. I am sure it is possible for a VM to be allocated a static address.

Reply:
> The problem here is that it changes every time - so you would have to update QGC every time. That's a real pain.

I know I have to deal with this all the time...

> I guess for now we document it, and think about possible alternatives.

Sure, it might be possible to automate this in some way. The only solution I know of so far is from @lyraik. He ran MAVLink router with the right configuration in WSL and that forwards the MAVLink to native Windows so it will auto connect. It's additional dependencies and would probably need to be scripted up but it's the easiest I know of.

> I am sure it is possible for a VM to be allocated a static address.

I don't think so, see https://superuser.com/a/1619390

Reply:
Im currently using  a .sh script to start everything:
with this command i get the ip of the wsl instance: 
`ip=$(tail -1 /etc/resolv.conf | cut -d' ' -f2)`
and then I just reroute everything:
`./mavlink-routerd -e $ip:14550 0.0.0.0:14550 & `

afterwards you can just start the simulator normaly in wsl and mavlink gets routed to windows

Reply:
Thanks @lyraik - interesting idea, but requires install/setup of additional dependencies. @MaEtUgR is aware of this and will doubtless evaluate in the coming weeks.

FWIW I think any setup where we only need to install one version of QGC and then everything just works is "a good thing".